includes/api/ApiQueryLogEvents.php

```diff
@@ -157,13 +153,26 @@ public function execute() {
 		$count = 0;
 		$res = $this->select( __METHOD__ );
 		$result = $this->getResult();
+
+		$userIds = $this->getFieldFromResults( $res, 'log_user' );
+		$userNames = $this->getIndexedUserNames( $userIds );
+
 		foreach ( $res as $row ) {
 			if ( ++ $count > $limit ) {
 				// We've reached the one extra which shows that there are additional pages to be had. Stop here...
 				$this->setContinueEnumParameter( 'start', wfTimestamp( TS_ISO_8601, $row->log_timestamp ) );
 				break;
 			}
 
+			if ( $this->fld_userid ) {
+				$row->user_id = $row->log_user;
+			}
+
+			if ( $this->fld_user ) {
+				$row->user_name = isset($userNames[$row->log_user]) ?
+					$userNames[$row->log_user] : 'unknown';
```

Comment:
I'd rather log a warning here. There should not be any mismatch of data. Returning `unknown` will not help in debugging :smiley: